core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3606,10 +3400,15 @@ object Stream extends StreamLowPriority {
     */
   def resourceWeak[F[_], O](r: Resource[F, O]): Stream[F, O] =
     r match {
-      case r: Resource.Allocate[F, O] =>
-        Stream.bracketCaseWeak(r.resource) { case ((_, release), e) => release(e) }.map(_._1)
-      case r: Resource.Bind[F, x, O] => resourceWeak(r.source).flatMap(o => resourceWeak(r.fs(o)))
-      case r: Resource.Suspend[F, O] => Stream.eval(r.resource).flatMap(resourceWeak)
+      case r: Resource.Allocate[f, o] =>
```

Comment:
This is a house of cards. The old version fails to compile on Dotty due to variance issues. Fixing those in a straightforward way via extractors breaks compilation on 2.12. This is cleanest version I could get compiling on all 3 versions.